ros_ws/src/query_services/query_services/new_object_client.py

```diff
@@ -0,0 +1,79 @@
+from capstone_interfaces.msg import StateObject, ObjectSpotted
+from capstone_interfaces.srv import AddObject
+from query_services import database_functions
+
+import sys
+
+import rclpy
+import time
+from numpy import uint32
+import builtin_interfaces
+from rclpy.node import Node
+
+
+class SpotNewObjectClient(Node):
+
+    def __init__(self):
+        super().__init__('spot_new_object_client')
+        self.cli = self.create_client(AddObject, 'add_new_object')
+        while not self.cli.wait_for_service(timeout_sec=1.0):
+            self.get_logger().info('service not available, waiting again...')
+        self.req = AddObject.Request()
+
+    def found_object(self, info: ObjectSpotted):# -> StateObject:
+        """
+        Add objects from given object information:
+        : param description
+        :return: all 
+        """
+        self.req.object_info.description = info.description
+        self.req.object_info.x = info.x
+        self.req.object_info.y = info.y
+        self.req.object_info.z = info.z
+        self.req.object_info.time_seen = info.time_seen
+
+        print(type(self.req))
+
+        print("TO REMOVE: SENDING REQUEST")
+        print(self.req)
+        print ("REQUEST SENT")
+        self.future = self.cli.call_async(self.req)
+        print("future",self.future.result())
+        print(self)
+        rclpy.spin_until_future_complete(self,self.future)
+        return self.future.result()
+
+
+def main():
+    rclpy.init()
+
+    spot_new_object = SpotNewObjectClient()
+
+    obj_info = ObjectSpotted()
```

Comment:
Why are we immediately creating and submitting an object when this service starts up? Is this debug code?

Reply:
This is the client side of it, so yes, running the client and submitting object info to the client assumes there is a new object to add. but taking in info via  input from keyboard was for debugging and demonstration vs how the system would actually function